### ISSUE TITLE
dataclass: Non-default argument after default argument

### ISSUE THREAD

**Issue by rayansostenes** [astral-sh/ty#1259]
### Summary

```python
from dataclasses import dataclass


@dataclass
class A:
    a: str | None = None
    b: str
    # The line above should produce and error, it will fail at runtime with:
    # >>> TypeError: non-default argument 'b' follows default argument
```
> https://play.ty.dev/5b8a6931-5da1-4d14-9db9-1b7a7bf5791b

### Version

ty 0.0.1-alpha.21 (ef52a1940 2025-09-19)

**Comment by sharkdp** [comment]
Thank you for reporting this. We already track this as a sub-item in #111.
